chain/test/Test/Pos/Chain/Block/Arbitrary.hs

```diff
@@ -451,12 +464,21 @@ genHeaderAndParams pm = do
             case header of
                 Block.BlockHeaderGenesis h -> h ^. Block.gbhExtra . Block.gehAttributes
                 Block.BlockHeaderMain h    -> h ^. Block.gbhExtra . Block.mehAttributes
+        thisEpochLeaderSchedule = nonEmpty (map Core.addressHash thisHeadersEpoch)
         params = Block.VerifyHeaderParams
             { Block.vhpPrevHeader = prev
             , Block.vhpCurrentSlot = randomSlotBeforeThisHeader
-            , Block.vhpLeaders = nonEmpty $ map Core.addressHash thisHeadersEpoch
+            , Block.vhpLeaders = case era of
+                Original         -> OriginalLeaders <$> thisEpochLeaderSchedule
+                OBFT ObftStrict  -> ObftStrictLeaders <$> thisEpochLeaderSchedule
+                OBFT ObftLenient -> case thisEpochLeaderSchedule of
+                    Nothing -> Nothing
+                    Just ls -> pure $ ObftLenientLeaders (Set.fromList (toList ls))
+                                                         (Core.BlockCount 5)
+                                                         (OldestFirst [])
```

Comment:
This generator, `genHeaderAndParams`, is actually critically flawed and it *always* returns a `vhpLeaders` which is `Nothing`. We're going to try to repair this test before merging CBR-482.

Reply:
> Instead of deconstructing & reconstructing the Maybe, we could use `do` notation:
> 
> ```
> OBFT ObftLenient -> do
>     ls <- thisEpochLeaderSchedule
>     pure $ ObftLenientLeaders (Set.fromList (toList ls))
>                               (Core.BlockCount 5)
>                               (OldestFirst [])
> ```
> but it's a minor difference.
Made suggested change